multi-arch-builders/builder-splunk.bu

```diff
@@ -0,0 +1,123 @@
+      user:
+        name: splunk
+      group:
+        name: splunk
+  files:
+    - path: /var/lib/systemd/linger/splunk
+      mode: 0644
+    - path: /home/splunk/.config/containers/systemd/splunk.container
+      mode: 0644
+      user:
+        name: splunk
+      group:
+        name: splunk
+      contents:
+        inline: |
+          [Unit]
+          Description=Start the splunk container to send journald logs to splunk
+          After=build-splunk.service
+          [Container]
+          ContainerName=rhcos-multiarch-splunk
+          Image=localhost/splunkforwarder-rhcos:latest
+          # set the user to root to run the splunk process
+          # as the splunk user on the host
+          User=root
+          Volume=/var/log/journal:/var/log/journal:ro
+          Volume=/run/log/journal:/run/log/journal:ro
+          Volume=/etc/machine-id:/etc/machine-id:ro
+          # use the container host's network to properly establish
+          # a connection with the splunk server
+          Network=host
+          # use `--ipc=host` to use the container host's IPC namespace
+          # to properly establish a connection with the splunk server
+          # use `--group-add keep-groups` to pass the systemd-journal group
+          # to access the journald logs inside the container
+          # user `--hostname=${SPLUNK_HOSTNAME}` to set a custom hostname
+          # to accurately index the journald logs in splunk.
+          PodmanArgs=--ipc=host --group-add keep-groups --hostname=${SPLUNK_HOSTNAME}
+          [Install]
+          WantedBy=default.target
+    - path: /home/splunk/.config/systemd/user/build-splunk.service
+      mode: 0644
+      user:
+        name: splunk
+      group:
+        name: splunk
+      contents:
+        inline: |
+          [Unit]
+          Description=Build the splunk container
+          [Service]
+          # Give time for the build to complete
+          TimeoutStartSec=10m
+          Type=oneshot
+          RemainAfterExit=yes
+          # Figure out soon if we can get the Red Hat Certs onto the hosts
+          Environment=GIT_SSL_NO_VERIFY=1
+          ExecStartPre=nm-online --timeout=30
+          # use `--arch=x86_64` because the RPM doesnt exist for aarch64.
+          # ppc64le and s390x RPMs are not compatible with journald input as mentioned
+          # in the splunk journald documentation for version 8.2.6
+          # https://docs.splunk.com/Documentation/Splunk/8.2.6/Data/CollecteventsfromJournalD
+          # The --arch=x86_64 works here because we have `qemu-user-static-x86`
+          # installed on non-x86_64 FCOS streams.
+          ExecStart=-podman build --pull-always --cache-ttl=480h --arch=x86_64 -t localhost/splunkforwarder:latest ${ITPAAS_SPLUNK_REPO}
+          ExecStart=-podman build --build-arg project=${DEPLOYMENT_CLIENT} --cache-ttl=480h --from localhost/splunkforwarder:latest -t localhost/splunkforwarder-rhcos ${SPLUNK_SIDECAR_REPO}
```

Comment:
feels like we don't really need to obfuscate the value of `DEPLOYMENT_CLIENT` do we?

Reply:
I'm not sure about this. It is a piece of the internal configuration that's used to establish a connection with the server.

Reply:
right, but it's just a string. I don't think there's really any need to hide it, but I'm fine with leaving it as it is.